### PR TITLE
[vtk-m] fix openmp on macos

### PR DESCRIPTION
**Pull request by autoantwort**
Fixes:
```
[56/343] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DVTKMDIY_DEBUG -DVTKMDIY_MPI_AS_LIB -DVTKMDIY_MPI_STATIC_BUILD -DVTKMDIY_NO_THREADS -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/arm64-osx-dbg/include -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/optionparser -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/diy -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/lcl/vtkmlcl -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/loguru -isystem /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/diy/vtkmdiy/include -isystem /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/arm64-osx-dbg/vtkm/thirdparty/diy/vtkmdiy/include/vtkmdiy/mpi -fPIC -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -fvisibility=hidden -Wno-pass-failed -Wall -Wcast-align -Wextra -Wpointer-arith -Wformat -Wformat-security -Wshadow -Wunused -fno-common -Wno-unused-function -Wconversion -Wodr -ffunction-sections "-Xclang -fopenmp" -std=c++14 -MD -MT vtkm/cont/CMakeFiles/vtkm_cont.dir/CellSetExplicit.cxx.o -MF vtkm/cont/CMakeFiles/vtkm_cont.dir/CellSetExplicit.cxx.o.d -o vtkm/cont/CMakeFiles/vtkm_cont.dir/CellSetExplicit.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/cont/CellSetExplicit.cxx
FAILED: vtkm/cont/CMakeFiles/vtkm_cont.dir/CellSetExplicit.cxx.o 
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -DVTKMDIY_DEBUG -DVTKMDIY_MPI_AS_LIB -DVTKMDIY_MPI_STATIC_BUILD -DVTKMDIY_NO_THREADS -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/arm64-osx-dbg/include -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/optionparser -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/diy -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/lcl/vtkmlcl -I/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/loguru -isystem /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/thirdparty/diy/vtkmdiy/include -isystem /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/arm64-osx-dbg/vtkm/thirdparty/diy/vtkmdiy/include/vtkmdiy/mpi -fPIC -g -arch arm64 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -fvisibility=hidden -Wno-pass-failed -Wall -Wcast-align -Wextra -Wpointer-arith -Wformat -Wformat-security -Wshadow -Wunused -fno-common -Wno-unused-function -Wconversion -Wodr -ffunction-sections "-Xclang -fopenmp" -std=c++14 -MD -MT vtkm/cont/CMakeFiles/vtkm_cont.dir/CellSetExplicit.cxx.o -MF vtkm/cont/CMakeFiles/vtkm_cont.dir/CellSetExplicit.cxx.o.d -o vtkm/cont/CMakeFiles/vtkm_cont.dir/CellSetExplicit.cxx.o -c /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/cont/CellSetExplicit.cxx
clang: warning: argument unused during compilation: '-Xclang -fopenmp' [-Wunused-command-line-argument]
In file included from /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/cont/CellSetExplicit.cxx:14:
In file included from /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/cont/Algorithm.h:16:
In file included from /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/cont/DeviceAdapter.h:20:
In file included from /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/cont/openmp/DeviceAdapterOpenMP.h:18:
In file included from /Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/cont/openmp/internal/DeviceAdapterAlgorithmOpenMP.h:19:
/Users/leanderSchulten/git_projekte/vcpkg/buildtrees/vtk-m/src/c0715241f1-a29aef6c50.clean/vtkm/cont/openmp/internal/FunctorsOpenMP.h:25:10: fatal error: 'omp.h' file not found
#include <omp.h>
         ^~~~~~~
1 error generated.
```

### CODE REVIEW

File: ports/vtk-m/omp.patch [hunk 1]
```diff
@@ -0,0 +1,26 @@
+diff --git a/CMake/VTKmDeviceAdapters.cmake b/CMake/VTKmDeviceAdapters.cmake
+index e9ac039..c745604 100644
+--- a/CMake/VTKmDeviceAdapters.cmake
++++ b/CMake/VTKmDeviceAdapters.cmake
+@@ -59,20 +59,14 @@ if(VTKm_ENABLE_OPENMP AND NOT TARGET vtkm::openmp)
+   find_package(OpenMP 4.0 REQUIRED COMPONENTS CXX QUIET)
+ 
+   add_library(vtkm::openmp INTERFACE IMPORTED GLOBAL)
++  target_link_libraries(vtkm::openmp INTERFACE OpenMP::OpenMP_CXX)
```
Comment (FrankXie05) [inline ports/vtk-m/omp.patch:9]: For using `OpenMP`, don't we need to pass `OpenMP_CXX_FLAGS`, `OpenMP_C_FLAGS`, `OpenMP_CXX_LIBRARIES`? 
The same doubt: https://github.com/microsoft/vcpkg/pull/30833

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: ports/vtk-m/omp.patch (hunk 1 above)
Comment (autoantwort) [reply ports/vtk-m/omp.patch:9]: > For using OpenMP, don't we need to pass OpenMP_CXX_FLAGS, OpenMP_C_FLAGS, OpenMP_CXX_LIBRARIES?

This is done by `target_link_libraries(vtkm::openmp INTERFACE OpenMP::OpenMP_CXX)`

File: ports/vtk-m/omp.patch [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+   if(OpenMP_CXX_FLAGS)
```
Comment (BillyONeal) [inline ports/vtk-m/omp.patch:10]: Should this also be off or...?

Comment (autoantwort) [reply ports/vtk-m/omp.patch:10]: `OpenMP_CXX_FLAGS` is populated, but the problem is that commands line arguments are separated by spaces. 
So`set_property(TARGET vtkm::openmp
 APPEND PROPERTY INTERFACE_COMPILE_OPTIONS $<$<COMPILE_LANGUAGE:CXX>:${OpenMP_CXX_FLAGS}>)` results in 
```
clang: warning: argument unused during compilation: '-Xclang -fopenmp'
```
If we directly use the target we don't have this problem.

Comment (autoantwort) [reply ports/vtk-m/omp.patch:10]: Inside the FindOpenMP module they use:
```cmake
        separate_arguments(_OpenMP_${LANG}_OPTIONS NATIVE_COMMAND "${OpenMP_${LANG}_FLAGS}")
        set_property(TARGET OpenMP::OpenMP_${LANG} PROPERTY
          INTERFACE_COMPILE_OPTIONS "$<$<COMPILE_LANGUAGE:${LANG}>:${_OpenMP_${LANG}_OPTIONS}>")
```

Comment (BillyONeal) [reply ports/vtk-m/omp.patch:10]: I see, you're saying it will always be on anyway so not messing with it is less patch. OK, I buy that.

**Review by BillyONeal — APPROVED** [review]

File: ports/vtk-m/omp.patch (hunk 2 above)
Comment (FrankXie05) [reply ports/vtk-m/omp.patch:10]: It seems that if we just want to link the `OpenMP` library, `cmake` will automatically use `OpenMP::OpenMP_CXX` as the link library to ensure that the `OpenMP` library can be found.

However, if we want the compiler to support `OpenMP`, we still need this `FLAGS`.

Here is enough for `vtk-m` to use `OpenMP`.

**Review by FrankXie05 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thank you!

**Pull request merged**